sdk/storage/Azure.Storage.Blobs/src/GlobalSuppressions.cs

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0004: DO provide both asynchronous and synchronous variants for all service methods.", Justification = "")]
```

Comment:
I don't think we should be adding GlobalSuppressions files to our client libraries -- removing them and removing suppressions is great, but adding new suppressions files feels like we are moving in the wrong direction.

Reply:
Unless we change AZC004 logic back, I don't think we have other solution.

Reply:
Would it be possible to share details regarding what changed in AZC004 and why a global suppression is needed vs. opting out of the warning for specific cases?

Reply:
I put it here. https://github.com/Azure/azure-sdk-tools/pull/6496#issue-1800304335
From "if there is an async method then it should have corresponding sync method" to "if there is an async method then it should have corresponding sync method, and if there is a sync method it should have corresponding async mthod".

Reply:
Why does the storage library have async methods without corresponding sync methods?  Are they valid cases?  I would like to understand why this was allowed for this library.  This seems like an important requirement that we shouldn't use analyzers to help detect for new additions in the future without having a good understanding of the reasons some methods were exempted.

Reply:
No they have sync methods without corresponding async methods, and I don't know why.